python_ta/contracts/__init__.py

```diff
@@ -1,40 +1,70 @@
 from typing import Any, List, Optional
 import typing
+import inspect
 import wrapt
 
 
-class InvalidPrecondition(Exception):
-    """Exception raised when precondition is invalid
+class InvalidCondition(Exception):
+    """Exception raised when a condition is invalid
     """
 
+
 @wrapt.decorator
-def check_contracts(wrapped, _instance, args, kwargs):
+def check_contracts(wrapped, instance, args, kwargs):
+    if instance is None and inspect.isclass(wrapped):
+        return add_class_contracts(wrapped, args, kwargs)
+    return check_function_contracts(wrapped, instance, args, kwargs)
```

Comment:
This is okay, but please use an `if`-`else` structure when the branches are roughly symmetric.

Reply:
Done